overlord/snapstate/target.go

```diff
@@ -1045,6 +1045,10 @@ func (p *updatePlan) validateAndFilterTargets(st *state.State, opts Options) err
 type UpdateGoal interface {
 	// toUpdate returns the data needed to update the snaps.
 	toUpdate(context.Context, *state.State, Options) (updatePlan, error)
+
+	// filterGatedSnaps validates the update plan against refresh control and, if
+	// applicable, filters out gated snaps.
+	filterGatedSnaps(*state.State, *updatePlan, Options) error
```

Comment:
I was unsure what to call this since it produces an error if we're installing specific snaps and filters without error if we're doing a general refresh. I ended up calling it `filter...` to make it evident that it might have side-effects on the plan

Reply:
thanks, Andrew that wrote this code originally might have input on naming